java/adapter/avro/src/main/java/org/apache/arrow/AvroToArrowUtils.java

```diff
@@ -0,0 +1,219 @@
+
+/**
+ * Class that does most of the work to convert Avro data into Arrow columnar format Vector objects.
+ */
+public class AvroToArrowUtils {
+
+  private static final int DEFAULT_BUFFER_SIZE = 256;
+
+  /**
+   * Creates an {@link org.apache.arrow.vector.types.pojo.ArrowType} from the {@link Schema.Field}
+   *
+   <p>This method currently performs following type mapping for Avro data types to corresponding Arrow data types.
+   *
+   * <ul>
+   *   <li>STRING --> ArrowType.Utf8</li>
+   *   <li>INT --> ArrowType.Int(32, signed)</li>
+   *   <li>LONG --> ArrowType.Int(64, signed)</li>
+   *   <li>FLOAT --> ArrowType.FloatingPoint(FloatingPointPrecision.SINGLE)</li>
+   *   <li>DOUBLE --> ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE)</li>
+   *   <li>BOOLEAN --> ArrowType.Bool</li>
+   *   <li>BYTES --> ArrowType.Binary</li>
+   * </ul>
+   */
+  private static ArrowType getArrowType(Type type) {
+
+    Preconditions.checkNotNull(type, "Avro type object can't be null");
+
+    final ArrowType arrowType;
+
+    switch (type) {
+      case STRING:
+        arrowType = new ArrowType.Utf8();
+        break;
+      case INT:
+        arrowType = new ArrowType.Int(32, /*signed=*/true);
+        break;
+      case BOOLEAN:
+        arrowType = new ArrowType.Bool();
+        break;
+      case LONG:
+        arrowType = new ArrowType.Int(64, /*signed=*/true);
+        break;
+      case FLOAT:
+        arrowType = new ArrowType.FloatingPoint(SINGLE);
+        break;
+      case DOUBLE:
+        arrowType = new ArrowType.FloatingPoint(DOUBLE);
+        break;
+      case BYTES:
+        arrowType = new ArrowType.Binary();
+        break;
+      default:
+        // no-op, shouldn't get here
+        throw new RuntimeException("Can't convert avro type %s to arrow type." + type.getName());
+    }
+
+    return arrowType;
+  }
+
+  /**
+   * Create Arrow {@link org.apache.arrow.vector.types.pojo.Schema} object for the given Avro {@link Schema}.
+   */
+  public static org.apache.arrow.vector.types.pojo.Schema avroToArrowSchema(Schema schema) {
+
+    Preconditions.checkNotNull(schema, "Avro Schema object can't be null");
+    List<Field> arrowFields = new ArrayList<>();
+
+    Schema.Type type = schema.getType();
+
+    if (type == Type.RECORD) {
+      throw new NotImplementedException();
+    } else if (type == Type.MAP) {
+      throw new NotImplementedException();
+    } else if (type == Type.UNION) {
+      throw new NotImplementedException();
+    } else if (type == Type.ARRAY) {
+      throw new NotImplementedException();
+    } else if (type == Type.ENUM) {
+      throw new NotImplementedException();
+    } else if (type == Type.NULL) {
+      throw new NotImplementedException();
+    } else {
+      final FieldType fieldType = new FieldType(true, getArrowType(type), null, null);
+      arrowFields.add(new Field("f0", fieldType, null));
+    }
+
+    return new org.apache.arrow.vector.types.pojo.Schema(arrowFields, null);
```

Comment:
please add a comment indicating what this parameter is.

Reply:
fixed.